### DISCUSSION TITLE
device goes offline and is caught in reconnect loop

### DISCUSSION THREAD

**Discussion by certabitmh** [aws/aws-iot-device-sdk-cpp-v2#744] [Q&A] [answered]
hi,

our devices to get disconnected from time to time. Especially at customers who turn off wifi over night, the device either never or several hours after wifi is available again comes online again.

I've no clue what the cause could be but have some log lines added below...appreciate any help (tweak iot params or other stuff?)

We are on release 1.29.1 

[log.txt](https://github.com/user-attachments/files/16549634/log.txt)

**Comment by bretambrose** [comment]
Log is unreadable, but fwiw, errno 101 is network unreachable (and this is the ipv4 connection attempt), which suggests that the problem is likely outside the SDK.

```
 [socket] - id=0xb3204f98 fd=11: connecting to endpoint 52.18.214.142:443."}] [4] tail.0: [1723117800.418554913, {"log"=>"[ERROR] [2024-08-08T11:49:30Z] [b3bff3b0] [socket] - id=0xb3204f98 fd=11: connect failed with error code 101."}]
```

**Reply by certabitmh** [reply]
thanks for fast response. changed log to txt attachment...

ok, understood, but

1. wifi connection is setup correct (besides that network dissapears at night and comes back in the morning), because I'm able to connect to device locally via laptop in same wifi network
2. all other customer online devices  work

is there any thread relevated stuff I could cause this myself?

**Comment by bretambrose** [comment, marked as answer]
Other than the connection failures (ipv6 unsupported, ipv4 host unreachable) the only unusual thing I notice is the fact there are 6 TLS context creation invocations in the logs when I would expect there to be none.  That would be something your code is doing.

**Reply by certabitmh** [reply]
thanks, will investigate this...

**Reply by certabitmh** [reply]
ok, this seems to improve it. interestingly sometime on re-establishing connection I get socket errors like 
`ping: can't create raw socket: Address family not supported by protocol`

In that case I've to disconnect and reconnect to wifi using wpa tools on linux. Then device connects normally again.

**But, a new issue came up now:**
Device is connected, but it seems subscription for shadow changes or job notifications are broken.
Only debug output on device I can see are pings/responses for it.

Is there a way to check if subscriptions are broken or to get notified?
